sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/orc/OrcFileFormat.scala

```diff
@@ -154,11 +155,17 @@ class OrcFileFormat
 
     (file: PartitionedFile) => {
       val conf = broadcastedConf.value.value
+      val metaCacheEnabled = conf.getBoolean(SQLConf.FILE_META_CACHE_ENABLED.key, false)
```

Comment:
It seems that there is no problem here. If `useFileMetaCacheList` config  is used, we can move the
```
val metaCacheEnabled = xxx
```
out of 

```
(file: PartitionedFile) => {
}
```